File: go/config/Statistic.go
package config

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// Statistic 对应 config/Statistic.py。
type Statistic struct {
	*session.ApiGroup
}

// NewStatistic 创建 Statistic API 分组。
func NewStatistic(s *session.Session) *Statistic {
	return &Statistic{ApiGroup: session.NewApiGroup(s)}
}

// Config 对应 statistic/config.xml。
func (s *Statistic) Config() (map[string]interface{}, error) {
	return s.S.Get("statistic/config.xml", nil, "config")
}

File: src/config/statistic.rs
//! Statistic config group (`config/Statistic.py`).

use serde_json::Value;

use crate::api::get_value;
use crate::connection::Connection;
use crate::errors::Result;

/// Statistic config group.
pub struct StatisticConfig<'a> {
    conn: &'a Connection,
}

impl<'a> StatisticConfig<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        StatisticConfig { conn }
    }

    /// `statistic/config.xml`. Statistic configuration.
    pub fn config(&self) -> Result<Value> {
        get_value(self.conn, "config/statistic/config.xml")
    }
}
